subsys/net/ip/net_if.c

```diff
@@ -4365,6 +4433,169 @@ bool net_if_is_promisc(struct net_if *iface)
 	return net_if_flag_is_set(iface, NET_IF_PROMISC);
 }
 
+int net_if_set_opt(struct net_if *iface, int optname, const void *optval, size_t optlen)
+{
+	int status = 0;
+
+	k_mutex_lock(&lock, K_FOREVER);
+
+	const struct net_if_api *api = net_if_get_device(iface)->api;
+
+	if (!(api->set_opt)) {
+		status = -ENOTSUP;
+		goto out;
+	}
+
+	/* Question for reviewers:
```

Comment:
In my opinion, global options should be handled by `net_if` itself. Since the current proposal makes a clear distinction between global/non-global options (with `NET_IF_GLOB_OPT_BASE`) it could be easily determined what should be forwarded down the inteface-specific implementation, and what not.